frontend/src/components/Dish/DishItem.jsx

```diff
@@ -137,21 +15,27 @@ function DishItem(props) {
   return (
     <ul style={listStyle}>
       <li>
-        <p>{dishState.description.english}</p>
+        <p>{dish.description[language]}</p>
       </li>
       <li>
-        <p>{dishState.name.english}</p>
+        <p>{dish.name[language]}</p>
       </li>
       <li>
-        <p>{dishState.price}</p>
+        <p>{dish.price}</p>
       </li>
       <li>
-        <img src={dishState.image.fileLink} alt="Raz" style={imagestyle}></img>
+        <img
+          src={dish.image ? dish.image.fileLink : ""}
+          alt="Raz"
+          style={imagestyle}
+        ></img>
       </li>
       <li>
         {isAdmin ? (
           <div>
-            <button onClick={preperForm}>{`${formType} Dish`}</button>
+            <button
+              onClick={() => preperForm("put", dish)}
+            >{`Edit Dish`}</button>
```

Comment:
I think you can have this edit/create button (along with the `isAdmin` logic) in the `Page`/`Navbar` generically

Reply:
And call it `Edit <entity>`, etc